draft-irtf-nwcrg-coding-and-congestion-01.txt

```diff
@@ -1,528 +1,448 @@
+   server).  It is worth pointing out that there are cases where these
+   channels are not separated.
+
+    Client                          Server
+
+   +------+                        +------+
+   |      | -- { data packet  -----|      |
+   |  CC  |                        |  CC  |
+   |      | -- received packet } --|      |
+   +------+                        +------+
+
+   +------+                        +------+
+   |      | -- { coded packet  ----|      |
+   | FEC  |                        | FEC  |
+   |      | -- repaired packet } --|      |
+   +------+                        +------+
+
+                 Figure 1: Notations and separate channels
+
+   Inside a host, the CC and FEC entities can be regarded as
+   conceptually separate:
+
+
+
+
+
+
+
+
+
+Kuhn, et al.              Expires July 22, 2020                 [Page 3]
+
+Internet-Draft            Coding and congestion             January 2020
+
+
+                 |                  --------                  ^
+                 | data             |      | data,            | sending
+                 |                  |      | requirements     | rate (or
+                 v                  |      v                  | window)
+  +-----------------------------+   |    +-------------------------+
+  |             FEC             | <----> |           CC            |
+  +-----------------------------+   |    +-------------------------+
+    ^                   |           |              ^
+    | information       | coded     |              | network
+    | about repaired    | data      |              | measurements
+    | ratio             -------------              |
+
+                        Figure 2: Separate entities
+
+   As Figure 2 shows, the inputs to FEC (data to work upon, and
+   signaling from the receiver about losses and/or repaired blocks) are
+   distinct from the inputs to CC.  The latter calculates a sending rate
+   or window from network measurements, and it takes the data to send as
+   input, sometimes along with application requirements such as upper/
+   lower rate bounds, periods of quiescence, or a priority.
+
+   It is not clear that the ACK signals feeding into a congestion
+   control algorithm are useful to FEC in their raw form, and vice versa
+   - information about repaired blocks may be quite irrelevant to a CC
+   algorithm.  However, there can be meaningful other interactions
+   (indicated by the horizontal double arrow) between the two entities,
+   usually as a result of their operation rather than by relaying their
+   own raw inputs.  For example, the network measurements carried out by
+   CC can yield a longer-term statistical measure such as a loss ratio
+   which is useful input for a FEC coding scheme.  Similarly, unequal
+   error protection using fountain codes can be used to assign different
+   priorities to blocks of data, and these priorities can be honored by
+   a CC mechanism.
```

Comment:
One of the thing after we have common grounds was to review a little bit this part. It should indeed present the different entities and say that they may be correlated. That being said, going further into providing examples may not be relevant - since examples will be given in other parts of the document. It was important to have this discussion when we agreed on the scope and this led to the layering approach - but we should simplify this version. 
Is that ok ?
Next commit will just comment the examples out.

Reply:
Yes, please mangle this paragraph as you see fit. Removing examples is fine.